plugin/ipythonPlugins/src/dist/iruby/iruby.js

```diff
@@ -27,267 +27,341 @@ define(function(require, exports, module) {
+            if (msg.status === "error")
+              finalStuff.status = "ERROR";
+            else
+              finalStuff.status = "FINISHED";
+  
+            if (!_.isEmpty(result) && finalStuff.payload === undefined) {
+              finalStuff.payload = "<pre>" + result + "</pre>";
+            }
           } else {
-            evaluation.outputdata.push( { type : 'err', value : content.data } );
+            var evaluation = { };
+            if (msg.status === "error")
+              evaluation.status = "ERROR";
+            else
+              evaluation.status = "FINISHED";
+            if (!_.isEmpty(result)) {
+              evaluation.payload = "<pre>" + result + "</pre>";
+            }
+            finalStuff = evaluation;
+            bkHelper.timeout(doFinish,250);
           }
-        } else {
-          var elem = $(document.createElement("div"));
-          var oa = new IPython.OutputArea(elem);
-          // twiddle the mime types? XXX
-          if (ipyVersion1) {
-            oa.append_mime_type(oa.convert_mime_types({}, content.data), elem, true);
+        }
+        var output = function output(a0, a1) {
+          if (_theCancelFunction === null || gotError)
+            return;
+          // this is called to write output
+          var type;
+          var content;
+          if (ipyVersion == '1') {
+            type = a0;
+            content = a1;
           } else {
-            oa.append_mime_type(content.data, elem);
+            type = a0.msg_type;
+            content = a0.content;
           }
-          var table = bkHelper.findTable(elem[0]);
-          if (table) {
-            evaluation.payload = table;
+
+          var evaluation = { };
+          evaluation.status = "RUNNING";
+
+          if ((ipyVersion == '3') ? (type === "error") : (type === "pyerr")) {
+            gotError = true;
+            var trace = _.reduce(content.traceback, function(memo, line) {
+              return  memo + "<br>" + myPython.utils.fixCarriageReturn(myPython.utils.fixConsole(line));
+            }, myPython.utils.fixConsole(content.evalue));
+
+            evaluation.payload = (content.ename === "KeyboardInterrupt") ?
+              "Interrupted" : [myPython.utils.fixConsole(content.evalue), trace];
+            if (finalStuff !== undefined) {
+              finalStuff.payload = evaluation.payload
+            }
+          } else if (type === "stream") {
+            evaluation.outputdata = [];
+            if (finalStuff !== undefined && finalStuff.outputdata !== undefined)
+              evaluation.outputdata = finalStuff.outputdata;
+            var text = (ipyVersion == '3') ? content.text : content.data;
+            evaluation.outputdata.push({type: (content.name === "stderr") ? 'err' : 'out',
+                                        value: text});
           } else {
-            evaluation.payload = elem.html();
+            var elem = $(document.createElement("div"));
```

Comment:
also this block here is redundant (from line 248 to 260): the same code is below in the else branch